src/proof/correct_key.rs

```diff
@@ -89,75 +86,77 @@ impl CorrectKey<EncryptionKey, DecryptionKey> for Paillier
             .map(|_| BigInt::sample_below(&ek.n))
             .collect();
 
-        let a : Vec<_> = r.par_iter()
-            .map(|ri| BigInt::modpow(ri, &ek.n, &ek.n))
+        let rn: Vec<_> = r.par_iter()
+            .map(|ri| BigInt::modpow(ri, &ek.n, &ek.nn))
             .collect();
 
         let e = compute_digest(
             iter::once(&ek.n)
-                .chain(&x)
-                .chain(&a)
+                .chain(&sn)
+                .chain(&rn)
         );
 
         let z: Vec<_> = r.par_iter()
-            .zip(y.par_iter())
-            .map(|(ri, yi)| (ri * BigInt::modpow(yi, &e, &ek.n)) % &ek.n)
+            .zip(s.par_iter())
+            .map(|(ri, si)| (ri * BigInt::modpow(si, &e, &ek.n)) % &ek.nn) // n or nn in the modpow?
```

Comment:
suppose to be `nn` in the `modpow`. (check https://pdfs.semanticscholar.org/802e/8b83755c11de06a57cc59a1810d4bdcc2da7.pdf?_ga=2.124342927.401960123.1532793082-1427010434.1526240291 section 4.2). 

It's interesting that the code passed benchmarks in this format - maybe I am missing something .